server/app/views/tags/IconElementTagProcessor.java

```diff
@@ -0,0 +1,44 @@
+package views.tags;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import controllers.AssetsFinder;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import org.thymeleaf.context.ITemplateContext;
+import org.thymeleaf.model.IProcessableElementTag;
+import org.thymeleaf.processor.element.AbstractElementTagProcessor;
+import org.thymeleaf.processor.element.IElementTagStructureHandler;
+import org.thymeleaf.templatemode.TemplateMode;
+
+public final class IconElementTagProcessor extends AbstractElementTagProcessor {
+  private final AssetsFinder assetsFinder;
+
+  public IconElementTagProcessor(String dialectPrefix, AssetsFinder assetsFinder) {
+    super(TemplateMode.HTML, dialectPrefix, "icon", true, null, false, 10);
+    this.assetsFinder = checkNotNull(assetsFinder);
+  }
+
+  @Override
+  protected void doProcess(
+      ITemplateContext context,
+      IProcessableElementTag tag,
+      IElementTagStructureHandler structureHandler) {
+    String html =
+        """
+            <img class="%s" src="%s" role="img" alt="" aria-hidden="true" />
+        """;
+
+    String type = tag.getAttributeValue("type");
+    String cssClass = tag.getAttributeValue("class");
+    String path = String.format("Images/uswds/icon-%s.svg", type);
+
+    if (Files.notExists(Paths.get("app/assets", path))) {
+      throw new RuntimeException(String.format("Can't find file: %s", path));
```

Comment:
Probably make this throw FileNotFoundException, despite it being a checked exception. 

However, I think I like the idea of creating an empty svg that can be a placeholder if the requested image isn't found and a very large error being logged that the file is missing.

On one hand I want to error hard early so devs can fix the mistakes, but on the other I don't want a missing image to break a production system.